### PR TITLE
Add a doc "Writing a plugin in C#"

### PR DESCRIPTION
**Pull request by WonyoungChoi**
- #333 
- Add a document "Writing a plugin in C#"

### CODE REVIEW

File: doc/develop-plugin-csharp.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+- `OnAttachedToEngine(IFlutterPluginBinding binding)`: This method is called when the plugin is registered to the `DotnetPluginRegistry` by the flutter application. In this method, you can create and initialize the platform channel.
+
+```c#
+        public void OnAttachedToEngine(IFlutterPluginBinding binding)
+        {
+            _channel = new MethodChannel("plugin_name", StandardMethodCodec.Instance, binding.BinaryMessenger);
+            _channel.SetMethodCallHandler(HandleMethodCall);
+        }
+```
```
Comment (swift-kim) [inline doc/develop-plugin-csharp.md:54]: Please indent properly.
```suggestion
  ```c#
  public void OnAttachedToEngine(IFlutterPluginBinding binding)
  {
      _channel = new MethodChannel("plugin_name", StandardMethodCodec.Instance, binding.BinaryMessenger);
      _channel.SetMethodCallHandler(HandleMethodCall);
  }
  ```
```

File: doc/develop-plugin-csharp.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        // Handle method calls on the plugin's channel asynchronously.
+        private async Task<object> HandleMethodCallAsync(MethodCall call) {
+            if (call.Method == "getPlatformVersion") {
+                return await Foo.GetPlatformVersionAsync();
+            }
+            throw new MissingPluginException();
+        }
+```
+
+In the callback method, you can use the `MethodCall` object to get the method name and arguments. If the method is completed successfully, just return any value to the application. If the method is failed, throw an exception to the application. The exception will be caught by the `MethodChannel` and the error message will be returned to the application. If the method is not implemented, throw a `MissingPluginException` to the application.
```
Comment (swift-kim) [inline doc/develop-plugin-csharp.md:86]: > If the method is failed, throw an exception to the application.

- "is failed" → "has failed" or "failed"
- Which type of exception should be thrown by the plugin?

File: doc/develop-plugin-csharp.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+
+### Available APIs
+
+Like a typical .NET library, the plugin package can be used .NET APIs and external NuGet packages. The following APIs are basically available in the plugin package:
```
Comment (swift-kim) [inline doc/develop-plugin-csharp.md:91]: "can be used"?

File: doc/develop-plugin-csharp.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+- **.NET Standard 2.0**: [.NET Standard 2.0](https://docs.microsoft.com/en-us/dotnet/api/?view=netstandard-2.0) is the default target framework for the plugin package. If you want to use other framework, you can change the `TargetFramework` in the `.csproj` file.
+- **Tizen .NET 4.0**: [Tizen .NET](https://github.com/Samsung/TizenFX) is a package that provides the Device APIs for Tizen. You can change the version of the package in the `.csproj` file.
+- **Embedder**: [Tizen.Flutter.Embedding](https://github.com/flutter-tizen/flutter-tizen/tree/master/embedding/csharp/Tizen.Flutter.Embedding) is a package that provides the APIs about plugin registration and communication with the flutter application. It is located in the `flutter-tizen/embedding/csharp` directory.
```
Comment (swift-kim) [inline doc/develop-plugin-csharp.md:95]: Embedder → Embedding

File: doc/develop-plugin.md [hunk 5]
```diff
@@ -9,11 +9,13 @@ Here are a few things you might consider when developing Flutter plugins for Tiz
 ### Implementation language
 
 - C/C++ (based on platform channels)
+- C# (based on platform channels)
 - Dart (based on Dart FFI)
 
 Typical Flutter plugins are written in their platform native languages, such as Java on Android and C/C++ on Tizen. However, some Windows plugins such as [`path_provider_windows`](https://github.com/flutter/plugins/tree/master/packages/path_provider/path_provider_windows) and Tizen plugins such as [`url_launcher_tizen`](https://github.com/flutter-tizen/plugins/tree/master/packages/url_launcher) are written in pure Dart using [Dart FFI](https://dart.dev/guides/libraries/c-interop) without any native code. To learn more about FFI-based plugins, you might read [Flutter Docs: Binding to native code using dart:ffi](https://flutter.dev/docs/development/platform-integration/c-interop).
 
-This document only covers native Tizen plugins written in C/C++.
+This document only covers native Tizen plugins written in C/C++. <br/>
```
Comment (swift-kim) [inline doc/develop-plugin.md:17]: `<br/>` isn't necessary.

File: doc/develop-plugin-csharp.md (hunk 3 above)
Comment (WonyoungChoi) [reply doc/develop-plugin-csharp.md]: OMG! Thanks!

File: doc/develop-plugin-csharp.md (hunk 1 above)
Comment (swift-kim) [reply doc/develop-plugin-csharp.md]: I mean, the entire code block should be indented with extra two spaces because it's part of the `<ul>` element.

Comment (WonyoungChoi) [reply doc/develop-plugin-csharp.md]: oh.. I understand now :)

File: doc/develop-plugin-csharp.md (hunk 2 above)
Comment (swift-kim) [reply doc/develop-plugin-csharp.md]: > Any exception type can be thrown, but `FlutterException` must be used to specify the code and details of the error.

Any exception type can be used but `FlutterException` must be used? Sounds like a contradiction.

Comment (WonyoungChoi) [reply doc/develop-plugin-csharp.md]: how about .. 
>  Any exception type can be thrown. However, if you want to specify the error code and details, use `FlutterException`.

?

Comment (swift-kim) [reply doc/develop-plugin-csharp.md]: Looks good :)

Comment (WonyoungChoi) [reply doc/develop-plugin-csharp.md]: Done. :smile_cat:

**Review by swift-kim — APPROVED** [review]
Overall looks good to me. Some additional language review might be required.

**Pull request merged**